spp_service_points/models/registrant.py

```diff
@@ -162,12 +170,41 @@ def _create_user(self, child_id):
                         (4, self.env.ref("base.group_user").id),
                         (4, self.env.ref("spp_service_points.service_point_users").id),
                     ],
-                    "is_service_point_user": True,
-                    "service_point_id": self.id,
+                    "service_point_ids": [(4, self.id)],
                 }
             )
         )
 
+    def get_user_id_list(self):
+        user_id_list = []
+
+        # get all user_ids of individual
+        if self.res_partner_company_id and self.res_partner_company_id.child_ids:
+            for child_id in self.res_partner_company_id.child_ids:
+                user_id_list.extend(child_id.user_ids.ids)
+
+        # to ensure no duplicate id
+        user_id_list = list(set(user_id_list))
+
+        return user_id_list
+
+    def write(self, vals):
+        res = super().write(vals)
+
+        if "res_partner_company_id" in vals:
+            self.user_ids = [(6, 0, self.get_user_id_list())]
+
+        return res
+
+    @api.model
+    def create(self, vals):
+        res = super().create(vals)
+
+        if "res_partner_company_id" in vals:
+            res.user_ids = [(6, 0, res.get_user_id_list())]
+
+        return res
```

Comment:
@reichie020212 if you have to override both write & create functions for adding logic to this field, I think its better to have a compute method instead